diff --git a/src/include/openssl/target.h b/src/include/openssl/target.h
index 4e12652..23bf5ee 100644
--- a/src/include/openssl/target.h
+++ b/src/include/openssl/target.h
@@ -40,6 +40,9 @@
 #elif defined(__MIPSEL__) && defined(__LP64__)
 #define OPENSSL_64_BIT
 #define OPENSSL_MIPS64
+#elif defined(__loongarch64)
+#define OPENSSL_64_BIT
+#define OPENSSL_LOONGARCH64
 #elif defined(__riscv) && __SIZEOF_POINTER__ == 8
 #define OPENSSL_64_BIT
 #define OPENSSL_RISCV64
