diff --git a/cpu/BUILD b/cpu/BUILD
index 6497452..e7afb92 100644
--- a/cpu/BUILD
+++ b/cpu/BUILD
@@ -188,3 +188,8 @@ constraint_value(
     name = "riscv64",
     constraint_setting = ":cpu",
 )
+
+constraint_value(
+    name = "loongarch64",
+    constraint_setting = ":cpu",
+)
diff --git a/host/extension.bzl b/host/extension.bzl
index 8756c59..ae4004d 100644
--- a/host/extension.bzl
+++ b/host/extension.bzl
@@ -15,6 +15,8 @@ def _translate_cpu(arch):
         return "mips64"
     if arch in ["riscv64"]:
         return "riscv64"
+    if arch in ["loongarch64"]:
+        return "loongarch64"
     return None
 
 def _translate_os(os):
